third_party/amd/lib/TritonAMDGPUTransforms/AccelerateAMDMatmul.cpp

```diff
@@ -511,23 +511,23 @@ class BlockedToMFMA : public OpRewritePattern<tt::DotOp> {
     }
 
     ttg::AMDMfmaEncodingAttr mfmaEnc = ttg::AMDMfmaEncodingAttr::get(
-        oldRetType.getContext(),
-        /*version*/ mfmaVersion, warpsPerTile, tilesPerWarp,
-        /*instrShape*/ mDim, nDim, /*isTransposed=*/isTransposed, CTALayout,
-        mfmaAccType);
+        oldRetType.getContext(), mfmaVersion, warpsPerTile, {mDim, nDim, kDim},
+        isTransposed, CTALayout, tilesPerWarp,
+        mfmaAccType.getIntOrFloatBitWidth());
 
     // convert accumulator
     auto oldAcc = dotOp.getC();
     auto newAcc = convertAndCastTensor(rewriter, oldAcc, mfmaEnc, mfmaAccType);
 
     // Here is a brief explanation of kWidth, kBase, and kDim
-    // 1. kWidth: the number of elements each thread loads from shared memory in
-    //    preparation for mfma instructions. In theory each thread can issue one
-    //    or more load instructions to load a total of kWidth elements, since
-    //    those elements are not required to be in contiguous addresses in
-    //    shared memory. But in practice, we make sure the kWidth elements can
-    //    be loaded from shared memory by a single ds_read instruction by
-    //    setting vecSize of the sharedLayout to be kWidth.
+    // 1. kWidth: the number of **consecutive** elements each thread loads from
+    //    shared memory in preparation for mfma instructions. In theory each
+    //    thread can issue one or more load instructions to load a total of
+    //    kWidth elements, since those elements are not required to be in
+    //    contiguous addresses in shared memory. But in practice, we make sure
```

Comment:
The "in theory..." sentense can still be confusing. I'd suggest to revise to: "One can issue multiple ds_read to load elements from non-contiguous addresses in shared memory for one mfma instruction, but that won't be good for performance. So in practice for better vectorization, we make sure.."